lib/p2p/Pool.ts

```diff
@@ -647,12 +642,26 @@ class Pool extends EventEmitter {
       this.pendingOutgoingConnections.delete(peer.nodePubKey!);
     });
 
-    peer.once('close', () => {
+    peer.once('close', async () => {
+      // cleanup
       if (peer.nodePubKey) {
         this.pendingOutgoingConnections.delete(peer.nodePubKey);
         this.peers.remove(peer.nodePubKey);
       }
       this.emit('peer.close', peer);
+
+      // if handshake passed and peer disconnected from us for stalling or without specifying any reason -
+      // reconnect, for that might have been due to a temporary loss in connectivity
+      const legitReconnect =
```

Comment:
`legitReconnect` seems a bit confusing, how about `shouldReconnect`?

Reply:
I see what you're saying. It's more like whether this disconnection happened deliberately, or due to temporary network issues. Maybe `unintentionalDisconnect` or `unintentionalClose` might be appropriate. But it's clear enough given the comments so I'm fine with whatever you prefer.